libcalico-go/lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -87,54 +108,75 @@ func (c *WorkloadEndpointClient) Delete(ctx context.Context, key model.Key, revi
 //
 // We store the IP addresses in annotations because patching the PodStatus directly races with changes that
 // kubelet makes so kubelet can undo our changes.
-func (c *WorkloadEndpointClient) patchInAnnotations(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+func (c *WorkloadEndpointClient) patchInAnnotations(ctx context.Context, kvp *model.KVPair, operation string) (*model.KVPair, error) {
+	var annotations map[string]string
+	var revision string
+	patchMode := PatchModeOf(ctx)
+	switch patchMode {
+	case PatchModeCNI:
+		annotations = c.calcCniAnnotations(kvp)
+		// Note: we drop the revision here because the CNI plugin can't handle a retry right now (and the kubelet
+		// ensures that only one CNI ADD for a given UID can be in progress).
+		revision = ""
+	default:
+		log.Errorf("no WorkloadEndpoint patching will be performed, unrecognised PatchMode passed to patchInAnnotations: %s.", patchMode)
```

Comment:
I don't think we should log an error here - lets leave that to the calling code.

Generally an error handling branch only needs to return an error or log an error, not both. Else you might get scenarios where an error is handled further up the stack but we still log out an error level log, which is confusing for the user.

Reply:
That makes sense, I'll remove it.